tests/functional/finetune/finetune_functional_test.py

```diff
@@ -82,6 +88,17 @@ def test_end2end_text_generation(run_input_map):
     finetune_model.delete()
 
 
+def test_cost_estimation_text_generation(estimate_cost_input_map):
+    model = ModelFactory.get(estimate_cost_input_map["model_id"])
+    dataset_list = [DatasetFactory.list(query=estimate_cost_input_map["dataset_name"])["results"][0]]
+    finetune = FinetuneFactory.create(str(uuid.uuid4()), dataset_list, model)
+    assert type(finetune.cost) is FinetuneCost
+    cost_map = finetune.cost.to_dict()
+    assert "trainingCost" in cost_map
+    assert "hostingCost" in cost_map
+    assert "inferenceCost" in cost_map
```

Comment:
Is there a way to assert the exact costs?

Reply:
No, as this is an estimate, the cost can vary depending on the random sample extracted from the dataset.